examples/dmd/parametric_tw_csv.cpp

```diff
@@ -223,14 +259,21 @@ int main(int argc, char *argv[])
         db = new CAROM::HDFDatabase();
     }
 
+    if (save_hdf)
+    {
+        hdf_db = new CAROM::HDFDatabase();
+        hdf_db->create("prediction" + to_string(myid) + ".hdf");
+    }
+
     string variable = string(var_name);
     int nelements = 0;
 
     if (csvFormat)
         db->getIntegerArray(string(data_dir) + "/dim.csv", &nelements, 1);
     else
     {
-        db->open(string(data_dir) + "/" + sim_name + "0/" + hdf_name, "r");
+        db->open(GetFilenameHDF(data_dir, sim_name, "0", hdf_name,
+                                myid, fileNameMode), "r");
         nelements = db->getDoubleArraySize("step0sol");
```

Comment:
This might be something for a future PR, but this should use `snap_pfx` and `snap_index_list` to be more general.  I've run into issues when I don't have a snapshot at 0.